### PR TITLE
Azure sharednetwork

### PR DESCRIPTION
**Pull request by Amoghrd**
Changes to QE workflow for sharednetwork on the basis of https://github.com/openshift/release/pull/26947

### CODE REVIEW

**Comment by Amoghrd** [comment]
@jianlinliu please review

File: ci-operator/step-registry/ipi/conf/azure/sharednetwork/ipi-conf-azure-sharednetwork-commands.sh [hunk 1]
```diff
@@ -5,20 +5,37 @@ set -o errexit
...
+  controlPlaneSubnet=$(cat ${SHARED_DIR}/customer_vnet_subnets.yaml | /tmp/yq '.platform.azure.controlPlaneSubnet')
+  computeSubnet=$(cat ${SHARED_DIR}/customer_vnet_subnets.yaml | /tmp/yq '.platform.azure.computeSubnet')
+  
+  PATCH="${SHARED_DIR}/install-config-qe-sharednetwork.yaml.patch"
 
-cat > "${PATCH}" << EOF
+  cat >> "${PATCH}" << EOF
+platform:
+  azure:
+    networkResourceGroupName: os4-common
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/azure/sharednetwork/ipi-conf-azure-sharednetwork-commands.sh:22]: `os4-common`? I think it should be the provisioned resource group where vent is created.

File: ci-operator/step-registry/ipi/conf/azure/sharednetwork/ipi-conf-azure-sharednetwork-commands.sh [hunk 2]
```diff
@@ -5,20 +5,37 @@ set -o errexit
...
 CONFIG="${SHARED_DIR}/install-config.yaml"
-PATCH=/tmp/install-config-sharednetwork.yaml.patch
 
-azure_region=$(/tmp/yq r "${CONFIG}" 'platform.azure.region')
+if [ -f "${SHARED_DIR}/customer_vnet_subnets.yaml" ]; then
+  vnet_name=$(cat ${SHARED_DIR}/customer_vnet_subnets.yaml | /tmp/yq '.platform.azure.virtualNetwork')
+  controlPlaneSubnet=$(cat ${SHARED_DIR}/customer_vnet_subnets.yaml | /tmp/yq '.platform.azure.controlPlaneSubnet')
+  computeSubnet=$(cat ${SHARED_DIR}/customer_vnet_subnets.yaml | /tmp/yq '.platform.azure.computeSubnet')
+  
+  PATCH="${SHARED_DIR}/install-config-qe-sharednetwork.yaml.patch"
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/azure/sharednetwork/ipi-conf-azure-sharednetwork-commands.sh:17]: How about naming it to `install-config-provisioned-sharednetwork.yaml.patch`

File: ci-operator/step-registry/ipi/conf/azure/sharednetwork/ipi-conf-azure-sharednetwork-commands.sh [hunk 3]
```diff
@@ -5,20 +5,37 @@ set -o errexit
...
   azure:
     networkResourceGroupName: os4-common
     virtualNetwork: do-not-delete-shared-vnet-${azure_region}
     controlPlaneSubnet: subnet-1
     computeSubnet: subnet-2
 EOF
-
-/tmp/yq m -x -i "${CONFIG}" "${PATCH}"
+fi
+/tmp/yq ea -i "${CONFIG}" "${PATCH}"
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/azure/sharednetwork/ipi-conf-azure-sharednetwork-commands.sh:41]: I run some local testing, seem like the command does not generate correct install-config:
```
[root@preserve-jialiu-ansible ~]# /tmp/yq --version
yq (https://github.com/mikefarah/yq/) version 4.22.1
[root@preserve-jialiu-ansible ~]# cat test.1 
platform:
  azure:
    virtualNetwork: 1.1
[root@preserve-jialiu-ansible ~]# cat test.2
platform:
  azure:
    networkResourceGroupName: os4-common
    virtualNetwork: do-not-delete-shared-vnet-${azure_region}
    controlPlaneSubnet: subnet-1
    computeSubnet: subnet-2
[root@preserve-jialiu-ansible ~]# /tmp/yq ea -i test.1 test.2
[root@preserve-jialiu-ansible ~]# cat test.1
platform:
  azure:
    virtualNetwork: 1.1
[root@preserve-jialiu-ansible ~]# cat test.2
platform:
  azure:
    virtualNetwork: 1.1
---
platform:
  azure:
    networkResourceGroupName: os4-common
    virtualNetwork: do-not-delete-shared-vnet-${azure_region}
    controlPlaneSubnet: subnet-1
    computeSubnet: subnet-2
```

Comment (jianlinliu) [reply ci-operator/step-registry/ipi/conf/azure/sharednetwork/ipi-conf-azure-sharednetwork-commands.sh:41]: The 3.3.0 ym command works well.
```
[root@preserve-jialiu-ansible ~]# yq --version
yq version 3.3.0
[root@preserve-jialiu-ansible ~]# cat test.1 
platform:
  azure:
    virtualNetwork: 1.1
[root@preserve-jialiu-ansible ~]# cat test.2
platform:
  azure:
    networkResourceGroupName: os4-common
    virtualNetwork: do-not-delete-shared-vnet-${azure_region}
    controlPlaneSubnet: subnet-1
    computeSubnet: subnet-2
[root@preserve-jialiu-ansible ~]# yq m -x -i test.1 test.2
[root@preserve-jialiu-ansible ~]# cat test.1
platform:
  azure:
    virtualNetwork: do-not-delete-shared-vnet-${azure_region}
    networkResourceGroupName: os4-common
    controlPlaneSubnet: subnet-1
    computeSubnet: subnet-2

```
Actually I prefer the 3.3.0 ym command, I can get most useful info via `--help` option, but I can not understand the help info from v4.22.1.

Comment (Amoghrd) [reply ci-operator/step-registry/ipi/conf/azure/sharednetwork/ipi-conf-azure-sharednetwork-commands.sh:41]: The reason I used v4.22.1 is for getting the vnet_name, computeSubnet and controlPlane subnet from the YAML file. The documentation is not good for 3.3..0 yq command and they have changed the command structure after 4.18 so I thought of using 4.22 there. But yeah will look into how to use 3.3.0 yq to read yaml files

**Comment by Amoghrd** [comment]
Rehearse passed

File: ci-operator/step-registry/ipi/conf/azure/sharednetwork/ipi-conf-azure-sharednetwork-commands.sh [hunk 4]
```diff
@@ -8,17 +8,35 @@ set -o pipefail
 curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -o /tmp/yq && chmod +x /tmp/yq
 
 CONFIG="${SHARED_DIR}/install-config.yaml"
-PATCH=/tmp/install-config-sharednetwork.yaml.patch
 
-azure_region=$(/tmp/yq r "${CONFIG}" 'platform.azure.region')
+if [ -f "${SHARED_DIR}/customer_vnet_subnets.yaml" ]; then
+  RESOURCE_GROUP=$(cat ${SHARED_DIR}/resouregroup)
+  vnet_name=$(cat ${SHARED_DIR}/customer_vnet_subnets.yaml | /tmp/yq r - '.platform.azure.virtualNetwork')
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/azure/sharednetwork/ipi-conf-azure-sharednetwork-commands.sh:14]: Just a minor suggestion, can we keep the same syntax as how the original code use `yq` load region from instlal-config.yaml? Something like `/tmp/yq r 'platform.azure.virtualNetwork' ${SHARED_DIR}/customer_vnet_subnets.yaml`.

Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/azure/sharednetwork/ipi-conf-azure-sharednetwork-commands.sh:14]: I do not think this is a correct syntax.
```
[root@preserve-jialiu-ansible ~]# cat test.2
platform:
  azure:
    networkResourceGroupName: os4-common
    virtualNetwork: do-not-delete-shared-vnet-${azure_region}
    controlPlaneSubnet: subnet-1
    computeSubnet: subnet-2
[root@preserve-jialiu-ansible ~]# cat test.2 | yq r - '.platform.azure.virtualNetwork'
[root@preserve-jialiu-ansible ~]# cat test.2 | yq r - 'platform.azure.virtualNetwork'
do-not-delete-shared-vnet-${azure_region}
```

And can we keep the same syntax as how the original code use yq to read region info from install-config.yaml? something like `yq r ${SHARED_DIR}/customer_vnet_subnets.yaml 'platform.azure.virtualNetwork'`

**Pull request merged**